cli/src/Morphir/Web/Graph/DependencyGraph.elm

```diff
@@ -35,42 +38,72 @@ viewGraph graph =
 dependencyGraph : SelectedModule -> Repo -> Element msg
 dependencyGraph selectedModule repo =
     let
-        gray =
-            rgb 0.9 0.9 0.9
-
-        filterDepsBySelectedModule : DAG.DAG FQName -> List ( String, List String )
-        filterDepsBySelectedModule deps =
+        filterDepsBySelectedModule : DAG.DAG FQName -> Maybe ModuleName -> List Dependency
+        filterDepsBySelectedModule deps moduleName =
+            let
+                createDependencyTuple nodeName childNodeNames =
+                    ( FQName.toString nodeName
+                    , Set.map FQName.toString childNodeNames
+                        |> Set.toList
+                    )
+            in
             deps
                 |> DAG.toList
-                |> List.filterMap
-                    (\( ( _, moduleName, _ ) as fqName, fqNameSet ) ->
-                        case selectedModule of
-                            Just ( _, selectedModName ) ->
-                                if selectedModName == moduleName then
-                                    Just
-                                        ( FQName.toString fqName
-                                        , Set.map FQName.toString fqNameSet
-                                            |> Set.toList
-                                        )
-
-                                else
-                                    Nothing
-
-                            Nothing ->
-                                Just
-                                    ( FQName.toString fqName
-                                    , Set.map FQName.toString fqNameSet
-                                        |> Set.toList
-                                    )
-                    )
+                |> (case moduleName of
+                        Just justmn ->
+                            List.filterMap
+                                (\( ( _, mn, _ ) as fqName, fqNameSet ) ->
+                                    if justmn == mn then
+                                        Just <| createDependencyTuple fqName fqNameSet
+
+                                    else
+                                        Nothing
+                                )
+
+                        Nothing ->
+                            List.map
+                                (\( fqName, fqNameSet ) ->
+                                    createDependencyTuple fqName fqNameSet
+                                )
+                   )
```

Comment:
This is very elegant but I think in this situation it's better to have a bit of duplication for readability. I would move the `deps |> DAG.toList` part inside both branches and move the pattern match to the top.